storm-client/src/jvm/org/apache/storm/daemon/supervisor/ClientSupervisorUtils.java

```diff
@@ -64,7 +64,7 @@ public static int processLauncherAndWait(Map<String, Object> conf, String user,
                                              final Map<String, String> environment, final String logPreFix)
         throws IOException {
         int ret = 0;
-        Process process = processLauncher(conf, user, null, args, environment, logPreFix, null, null);
+        Process process = processLauncher(conf, user, null, args, environment, null, null, null);
         if (StringUtils.isNotBlank(logPreFix)) {
```

Comment:
Why not remove this here instead for consistency? I just searched all the callers and all except here rely on `processLauncher` and don't call `Utils.readAndLogStream`.

Reply:
I think the difference is that `processLauncherAndWait` wait for the process to finish, which is synchronous.  `processLauncher` doesn't wait, which is asynchronous.

Reply:
I meant do we really need to handle `Utils.readAndLogStream` here instead of deferring in `processLauncher`? If not we can just remove the if statement instead.

Reply:
OK I got it. You're right. What we should do it is STORM-3389 instead, the change here looks reasonable.